src/DynamoCoreWpf/Views/Core/WorkspaceView.xaml.cs

```diff
@@ -182,6 +184,67 @@ internal Rect GetVisibleBounds()
             return new Rect(topLeft, bottomRight);
         }
 
+        internal void SaveWorkspaceAsImage(string path)
+        {
+            var initialized = false;
+            var bounds = new Rect();
+
+            var dragCanvas = WpfUtilities.ChildOfType<DragCanvas>(this);
+            var childrenCount = VisualTreeHelper.GetChildrenCount(dragCanvas);
+            for (int index = 0; index < childrenCount; ++index)
+            {
+                var child = VisualTreeHelper.GetChild(dragCanvas, index);
+                var firstChild = VisualTreeHelper.GetChild(child, 0);
+                if ((!(firstChild is NodeView)) && (!(firstChild is NoteView)) && (!(firstChild is AnnotationView)))
+                    continue;
+
+                var childBounds = VisualTreeHelper.GetDescendantBounds(child as Visual);
+                childBounds.X = (double)(child as Visual).GetValue(Canvas.LeftProperty);
+                childBounds.Y = (double)(child as Visual).GetValue(Canvas.TopProperty);
+
+                if (initialized)
+                {
+                    bounds.Union(childBounds);
+                }
+                else
+                {
+                    initialized = true;
+                    bounds = childBounds;
+                }
+            }
+
+            // Nothing found in the canvas, bail out.
+            if (!initialized) return;
```

Comment:
Addressed review comment. Note that even without this `return`, the `Math.Ceiling` below turns out to work: it makes the final PNG to be of `20 x 20px` in dimension.